### PR TITLE
Add scalafmt and format sources

### PR DESCRIPTION
**Pull request by jto**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,35 +1,40 @@
...
+    publishTo := {
+      val nexus = "https://oss.sonatype.org/"
+      if (version.value.endsWith("-SNAPSHOT"))
+        Some("snapshots" at nexus + "content/repositories/snapshots")
+      else
+        Some("releases" at nexus + "service/local/staging/deploy/maven2")
+    },
+    pomExtra := (
+      <developers>
       <developer>
```
Comment (sjrd) [inline build.sbt]: This looks like it's actually misindented. :-s

**Comment by sjrd** [comment]
If we do that, we should also definitely add `scalafmtCheck` to the CI.

File: src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala [hunk 2]
```diff
@@ -42,12 +46,15 @@ object MissingLinkPlugin extends AutoPlugin {
...
+      inConfig(Test)(configSettings)
   }
 
-  private def loadArtifactsAndCheckConflicts(cp: Seq[File],
-      classDirectory: File, log: Logger): Seq[Conflict] = {
+  private def loadArtifactsAndCheckConflicts(
+    cp: Seq[File],
+    classDirectory: File,
+    log: Logger
+  ): Seq[Conflict] = {
```
[the remark this answers is not in the document]
Comment (jto) [reply src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala]: No really. [scalafmt can't do that safely](https://github.com/scalameta/scalafmt/blob/master/scalafmt-core/shared/src/main/scala/org/scalafmt/config/BinPack.scala#L7-L16)

**Comment by jto** [comment]
@sjrd fixed :)

File: build.sbt [hunk 3]
```diff
@@ -1,35 +1,42 @@
...
+          <id>sjrd</id>
+          <name>Sébastien Doeraene</name>
+          <url>https://github.com/sjrd/</url>
+        </developer>
+      </developers>
+      // format: on
+    ),
+    pomIncludeRepository := { _ =>
+      false
+    }
```
Comment (sjrd) [inline build.sbt]: Could you restore the trailing comma here, please?

File: src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (sjrd) [reply src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala]: That is really annoying. I'll let it go in the interest of getting scalafmt in for this project, but I don't like it.

File: src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala [hunk 4]
```diff
@@ -137,7 +147,7 @@ object MissingLinkPlugin extends AutoPlugin {
 
     val descriptions = Map(
       ConflictCategory.CLASS_NOT_FOUND -> "Class being called not found",
-      ConflictCategory.METHOD_SIGNATURE_NOT_FOUND -> "Method being called not found",
+      ConflictCategory.METHOD_SIGNATURE_NOT_FOUND -> "Method being called not found"
```
Comment (sjrd) [inline src/main/scala/ch/epfl/scala/sbtmissinglink/MissingLinkPlugin.scala]: Could you restore the trailing comma here, please?

**Pull request merged**